python/tank/bootstrap/manager.py

```diff
@@ -57,25 +60,57 @@ def __init__(self, sg_user=None):
         # defaults
         self._bundle_cache_fallback_paths = []
         self._caching_policy = self.CACHE_SPARSE
-        self._pipeline_configuration_name = None
+        self._pipeline_configuration_identifier = None
         self._base_config_descriptor = None
         self._progress_cb = None
         self._do_shotgun_config_lookup = True
         self._plugin_id = None
 
         log.debug("%s instantiated" % self)
 
-
     def __repr__(self):
+        if self._pipeline_configuration_identifier is None:
+            identifier_type = "is"
+        elif isinstance(self._pipeline_configuration_identifier, int):
+            identifier_type = "id"
+        else:
+            identifier_type = "name"
+
         repr  = "<TkManager "
         repr += " User %s\n" % self._sg_user
         repr += " Cache fallback path %s\n" % self._bundle_cache_fallback_paths
         repr += " Caching policy %s\n" % self._caching_policy
         repr += " Plugin id %s\n" % self._plugin_id
-        repr += " Config %s\n" % self._pipeline_configuration_name
+        repr += " Config %s %s\n" % (identifier_type, self._pipeline_configuration_identifier)
         repr += " Base %s >" % self._base_config_descriptor
         return repr
 
+    def enumerate_pipeline_configurations(self, project=None):
```

Comment:
FWIW I think we should use generators more often in our code. The code is simpler to write and for the receiving end it only gets iterated on one. I'll switch this to a list since its not Toolkit-like in its approach/